test/e2e/operator_test.go

```diff
@@ -886,3 +942,43 @@ func tryRemovingTaintWithExpectedSuccess(ctx context.Context, t *testing.T, clie
 func tryRemovingTaintWithExpectedFailure(ctx context.Context, t *testing.T, clientSet *k8sclient.Clientset, node *corev1.Node, taint *corev1.Taint) {
 	tryUpdatingTaintWithExpectation(ctx, t, clientSet, node, taint, false, false)
 }
+
+func applyFakePSIVolume(ctx context.Context, kubeClient *k8sclient.Clientset) error {
+	operatorDeployment, err := kubeClient.AppsV1().Deployments(operatorclient.OperatorNamespace).Get(ctx, operatorclient.OperandName+"-operator", metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	volume := corev1.Volume{
+		Name: fakePSIVolume,
+		VolumeSource: corev1.VolumeSource{
+			EmptyDir: &corev1.EmptyDirVolumeSource{},
+		},
+	}
+	volumeMount := corev1.VolumeMount{
+		Name:      fakePSIVolume,
+		MountPath: "/proc/pressure",
+	}
+	operatorDeployment.Spec.Template.Spec.Volumes = append(operatorDeployment.Spec.Template.Spec.Volumes, volume)
+	operatorDeployment.Spec.Template.Spec.Containers[0].VolumeMounts = append(operatorDeployment.Spec.Template.Spec.Containers[0].VolumeMounts, volumeMount)
+	return updateDeploymentAndRetryOnConflicts(ctx, kubeClient, operatorDeployment, metav1.UpdateOptions{})
+}
+
+func dropFakePSIVolume(ctx context.Context, kubeClient *k8sclient.Clientset) error {
+	operatorDeployment, err := kubeClient.AppsV1().Deployments(operatorclient.OperatorNamespace).Get(ctx, operatorclient.OperandName+"-operator", metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	var volumes []corev1.Volume
+	var volumeMounts []corev1.VolumeMount
+	for _, v := range operatorDeployment.Spec.Template.Spec.Volumes {
+		if v.Name != fakePSIVolume {
+			volumes = append(volumes, v)
+		}
+	}
+	for _, vm := range operatorDeployment.Spec.Template.Spec.Containers[0].VolumeMounts {
+		if vm.Name != fakePSIVolume {
+			volumeMounts = append(volumeMounts, vm)
+		}
+	}
+	return updateDeploymentAndRetryOnConflicts(ctx, kubeClient, operatorDeployment, metav1.UpdateOptions{})
```

Comment:
Neither of `volumes` and `volumeMounts` seems to be put back to `operatorDeployment`.

Reply:
Unfortunately we can't really mount anything under `/proc`.
Let's use an env variable to override it for testing purposes.